test/suites/test_push.py

```diff
@@ -0,0 +1,449 @@
+
+    def test_00_06_push_delete_via_connect(self):
+        def assert_push_delete(conn):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                conn.insert('tester', (1,'Bill'))
+                resp = conn.delete(
+                    'tester',
+                    1,
+                    on_push=push_callback,
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[100, 1]], [[200, 1]], [[300, 1]]])
+            self.assertEqual(resp.data[0], [1, 'Bill'])
+
+        conn = tarantool.connect(host=self.host, port=self.port, user='test', password='test')
+        # Check push from instance.
+        assert_push_delete(conn)
+        conn.close()
+        # Stop instance.
+        self.srv.stop()
+
+        with self.assertRaises(NetworkError):
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                conn.ping()
+
+    @skip_or_run_push_test_via_conn_pool
+    def test_01_00_push_call_via_connection_pool(self):
+        def assert_push_call(conn_pool):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                resp = conn_pool.call(
+                    'server_function',
+                    mode=tarantool.Mode.PREFER_RO,
+                    on_push=push_callback, 
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[1, 1]], [[2, 1]], [[3, 1]]])
+            self.assertEqual(resp.data[0], [3, 0])
+
+        conn_pool = tarantool.ConnectionPool([{'host':self.host, 'port':self.port}], user='test', password='test')
+        # Check push from instance.
+        assert_push_call(conn_pool)
+        conn_pool.close()
+        # Stop instance.
+        self.srv.stop()
+
+    @skip_or_run_push_test_via_conn_pool
+    def test_01_01_push_eval_via_connection_pool(self):
+        def assert_push_eval(conn_pool):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                resp = conn_pool.eval(
+                    'return server_function()',
+                    mode=tarantool.Mode.PREFER_RO,
+                    on_push=push_callback, 
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[1, 1]], [[2, 1]], [[3, 1]]])
+            self.assertEqual(resp.data[0], [3, 0])
+
+        conn_pool = tarantool.ConnectionPool([{'host':self.host, 'port':self.port}], user='test', password='test')
+        # Check push from instance.
+        assert_push_eval(conn_pool)
+        conn_pool.close()
+        # Stop instance.
+        self.srv.stop()
+
+    @skip_or_run_push_test_via_conn_pool
```

Comment:
It's like you can do a test matrix here (I couldn't say that there are good examples of it, but at least there is some working examples, like in decimal/datetime)

Reply:
Done, implemented test cases inside `push_test_cases`